transaction/examples/resources/creation/fungible/no_initial_supply.rtm

```diff
@@ -16,52 +16,36 @@ CALL_METHOD
-                    Enum<Option::Some>(Enum<AccessRule::DenyAll>()),            # Mintoer Updater Role Rule
-                    true                                                        # Temporary boolean, ignore as this will be removed soon
-                )
+                Some(Enum<AccessRule::AllowAll>()),  # Minter (if None: defaults to Owner)
+                Some(Enum<AccessRule::DenyAll>())    # Minter Updater (if None: defaults to Owner)
             )
         ),
-        Enum<0u8>(),                                                            # Use Default Burn Roles
-        Enum<0u8>(),                                                            # Use Default Freeze Roles
-        Enum<0u8>(),                                                            # Use Default Recall Roles
-        Enum<0u8>(),                                                            # Use Default Withdraw Roles
-        Enum<0u8>()                                                             # Use Default Deposit Roles
+        None,        # Burn Roles (if None: defaults to DenyAll, DenyAll)
+        None,        # Freeze Roles (if None: defaults to DenyAll, DenyAll)
+        None,        # Recall Roles (if None: defaults to DenyAll, DenyAll)
+        None,        # Withdraw Roles (if None: defaults to AllowAll, DenyAll)
+        None         # Deposit Roles (if None: defaults to AllowAll, DenyAll)
     )
-    Tuple(
-        Map<String, Tuple>(
+    Tuple(                                                                   # Metadata initialization
+        Map<String, Tuple>(                                                  # Initial metadata values
             "name" => Tuple(
-                Enum<Option::Some>(
-                    Enum<Metadata::String>("MyResource")                               # Resource Name
-                ),
-                true                                                                   # Locked Resource Name
-            ),
-            "symbol" => Tuple(
-                Enum<Option::Some>(
-                    Enum<Metadata::String>("RSRC"),                                   # Resource Symbol
-                ),
-                true                                                                   # Locked Resource Symbol
-            ),
-            "description" => Tuple(
-                Enum<Option::Some>(
-                    Enum<Metadata::String>("A very innovative and important resource")    # Resource Description
-                ),
-                true                                                                      # Locked Resource Description
+                Some(Enum<Metadata::String>("MyResource")),    # Resource Name
+                true                                                         # Locked
             )
         ),
-        Map<String, Tuple>()
+        Map<String, Enum>(                                                   # Metadata roles
```

Comment:
Not something we need to change - just an observation - that this is inconsistent with the tuple structure above. I guess it makes sense, because they have different defaulting strategies.

Reply:
Yup, exactly